### PR TITLE
MAGECLOUD-3196: Add ElasticSearch 6.x Support to Cloud

### PR DESCRIPTION
**Pull request by oshmyheliuk**
### Description
Added ES 6.x support to Cloud 

### Fixed Issues (if relevant)

https://magento2.atlassian.net/browse/MAGECLOUD-3196

### Manual testing scenarios

https://jira.corp.magento.com/browse/MAGECLOUD-57

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: src/Test/Unit/Process/Deploy/InstallUpdate/ConfigUpdate/SearchEngine/ConfigTest.php [hunk 1]
```diff
@@ -246,9 +241,22 @@ public function testGetWithElasticSearchDataProvider(): array
...
-                    'elasticsearch5_server_hostname' => 'localhost',
-                    'elasticsearch5_server_port' => 1234,
+                    'engine' => 'elasticsearch6',
+                    'elasticsearch6_server_hostname' => 'localhost',
+                    'elasticsearch6_server_port' => 1234,
+                ],
+            ],
+            [
+                'customSearchConfig' => [],
+                'version' => '7.2',
```
Comment (NadiyaS) [inline src/Test/Unit/Process/Deploy/InstallUpdate/ConfigUpdate/SearchEngine/ConfigTest.php]: this dataprovider checks only > sign in version checking, you need to check and equality too

File: src/Process/Deploy/InstallUpdate/ConfigUpdate/SearchEngine/Config.php [hunk 2]
```diff
@@ -124,7 +124,10 @@ private function getElasticSearchConfiguration(array $config)
     {
         $engine = 'elasticsearch';
 
-        if (Semver::satisfies($this->elasticSearch->getVersion(), '>= 5')) {
+        $esVersion = $this->elasticSearch->getVersion();
+        if (Semver::satisfies($esVersion, '>= 6')) {
```
Comment (NadiyaS) [inline src/Process/Deploy/InstallUpdate/ConfigUpdate/SearchEngine/Config.php]: just for the future. If tendency is the same, we just need to retrieve major version for engine name which is higher v.5

File: src/Test/Unit/Process/Deploy/InstallUpdate/ConfigUpdate/SearchEngine/ConfigTest.php (hunk 1 above)
Comment (oshmyheliuk) [reply src/Test/Unit/Process/Deploy/InstallUpdate/ConfigUpdate/SearchEngine/ConfigTest.php]: Done

**Review by NadiyaS — APPROVED** [review]

**Review by NadiyaS — APPROVED** [review]

File: src/Process/Deploy/InstallUpdate/ConfigUpdate/SearchEngine/Config.php (hunk 2 above)
Comment (oshmyheliuk) [reply src/Process/Deploy/InstallUpdate/ConfigUpdate/SearchEngine/Config.php]: Nice catch, changed to this logic

**Comment by mveeramneni** [comment]
Cannot access products on frontend page. Encountered following error report 
`{"0":"Undefined factory elasticsearch6","1":"#0 \/app\/vendor\/magento\/module-catalog\/Model\/Layer.php(146): Magento\\Elasticsearch\\Model\\Layer\\Category\\ItemCollectionProvider->getCollection(Object(Magento\\Catalog\\Model\\Category\\Interceptor))\n#1 \/app\/vendor\/magento\/module-catalog\/Block\/Product\/ListProduct.php(470): Magento\\Catalog\\Model\\Layer->getProductCollection()\n#2 \/app\/vendor\/magento\/module-catalog\/Block\/Product\/ListProduct.php(116): Magento\\Catalog\\Block\\Product\\ListProduct->initializeProductCollection()\n#3 \/app\/vendor\/magento\/module-catalog\/Block\/Product\/ListProduct.php(357): Magento\\Catalog\\Block\\Product\\ListProduct->_getProductCollection()\n#4 \/app\/vendor\/magento\/module-page-cache\/Model\/Layout\/LayoutPlugin.php(71): Magento\\Catalog\\Block\\Product\\ListProduct->getIdentities()\n#5 \/app\/vendor\/magento\/framework\/Interception\/Interceptor.php(146): Magento\\PageCache\\Model\\Layout\\LayoutPlugin->afterGetOutput(Object(Magento\\Framework\\View\\Layout\\Interceptor), '\\n<script>\\n    r...')\n#6 \/app\/vendor\/magento\/framework\/Interception\/Interceptor.php(153): Magento\\Framework\\View\\Layout\\Interceptor->Magento\\Framework\\Interception\\{closure}()\n#7 \/app\/generated\/code\/Magento\/Framework\/View\/Layout\/Interceptor.php(52): Magento\\Framework\\View\\Layout\\Interceptor->___callPlugins('getOutput', Array, Array)\n#8 \/app\/vendor\/magento\/framework\/View\/Result\/Page.php(257): Magento\\Framework\\View\\Layout\\Interceptor->getOutput()\n#9 \/app\/vendor\/magento\/framework\/View\/Result\/Layout.php(170): Magento\\Framework\\View\\Result\\Page->render(Object(Magento\\Framework\\App\\Response\\Http\\Interceptor))\n#10 \/app\/vendor\/magento\/framework\/Interception\/Interceptor.php(58): Magento\\Framework\\View\\Result\\Layout->renderResult(Object(Magento\\Framework\\App\\Response\\Http\\Interceptor))\n#11 \/app\/vendor\/magento\/framework\/Interception\/Interceptor.php(138): Magento\\Framework\\View\\Result\\Page\\Interceptor->___callParent('renderResult', Array)\n#12 \/app\/vendor\/magento\/framework\/Interception\/Interceptor.php(153): Magento\\Framework\\View\\Result\\Page\\Interceptor->Magento\\Framework\\Interception\\{closure}(Object(Magento\\Framework\\App\\Response\\Http\\Interceptor))\n#13 \/app\/generated\/code\/Magento\/Framework\/View\/Result\/Page\/Interceptor.php(26): Magento\\Framework\\View\\Result\\Page\\Interceptor->___callPlugins('renderResult', Array, Array)\n#14 \/app\/vendor\/magento\/framework\/App\/Http.php(139): Magento\\Framework\\View\\Result\\Page\\Interceptor->renderResult(Object(Magento\\Framework\\App\\Response\\Http\\Interceptor))\n#15 \/app\/vendor\/magento\/framework\/App\/Bootstrap.php(257): Magento\\Framework\\App\\Http->launch()\n#16 \/app\/pub\/index.php(40): Magento\\Framework\\App\\Bootstrap->run(Object(Magento\\Framework\\App\\Http\\Interceptor))\n#17 {main}","url":"\/index.php\/cat-1.html","script_name":"\/index.php"}`

**Comment by mveeramneni** [comment]
default ES configuration works but there are issues with custom configurations. May be it's a problem with ece-tools. https://jira.corp.magento.com/browse/MAGECLOUD-57 test fails but approving PR as default config works.

**Comment by mveeramneni** [comment]
QA Approved

**Pull request merged**